### ISSUE TITLE
How to list files to operate

### ISSUE THREAD

**Issue by zelon** [Azure/azure-storage-azcopy#424]
##### Note: The version is visible when running AzCopy without any argument
10.1.2

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.
azcopy copy/sync
### What problem was encountered?
I cannot find the /L option of previous version. The '/L option' was used to list files to operate
### How can we reproduce the problem in the simplest way?
run azcopy copy/sync
### Have you found a mitigation/solution?
no

**Comment by zezha-msft** [comment]
Hi @zelon, thanks for reaching out!

To clarify, are we talking about a dry-run flag? For example, the files to be transferred are listed first without doing any actual work.

If yes, we have this feature on our backlog and will get to it shortly. If not, please let us know. Thanks!

**Comment by zelon** [comment]
yes, I want the 'dry-run' flag. I'm looking forward to the work. Thanks!

**Comment by zezha-msft** [comment]
@zelon sounds good! We'll update this thread once the work is scheduled.

**Comment by Fuco1** [comment]
Especially for `rm` with include / exclude flags this seems like a must-have feature.

**Comment by eoq** [comment]
Yes. Especially for the sync operation. Ideally the results stored in the log file would be easily parsable so that the list of adds/deletes/overwrites could be pulled out and used pro grammatically if required
Thanks
Ed

**Comment by Fuco1** [comment]
Hey, any progress on this front?  It's still relevant for people in my company, we've wiped some data a bunch of times :blush: (so far we had backups, but...)

**Comment by adreed-msft** [comment]
Currently this is not a primary focus for us, but it is still on our radar. I'm sorry to hear that the lack of it is continuing to cause you trouble.

**Comment by asears** [comment]
This would be an important story for backup and disaster recovery scenarios.  --dry-run should be available for testing prior to running in an environment to prevent data loss.

**Comment by hoshy** [comment]
> Currently this is not a primary focus for us, but it is still on our radar. I'm sorry to hear that the lack of it is continuing to cause you trouble.

Any update on this?

**Comment by zezha-msft** [comment]
Sorry for the delay, I will bring this issue up for prioritization with the team. Thanks.

**Comment by mahalel** [comment]
Hi, this would be a great feature to have, similar to how rclone --dry-run flag works - good for testing before writing a new script.

**Comment by olastor** [comment]
How can it be that there seems to be no `--dryrun` flag for copying files and this issue is open since almost three years? You'd assume such basic functionality is included from the very beginning.

**Comment by espenairmine** [comment]
Hi, any update on this? What was the feedback from the team when this was brought up in December?
